### PR TITLE
Add reference resolver for target_id for aws_lb_target_group_attachment

### PR DESCRIPTION
**Pull request by smcavallo**
### Description of your changes

Add reference resolver for `target_id` for `aws_lb_target_group_attachment`
See - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_target_group_attachment

This is another case where the `target_id` could resolve to multiple resource types.  The most common use case though is most likely ALB.

Fixes #

I have:

- [X ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
local e2e

### CODE REVIEW

**Comment by sergenyalcin** [comment]
We may consider fixing the `examples/elbv2/lbtargetgroupattachment.yaml` and triggering the automated tests.

**Comment by smcavallo** [comment]
Hi @sergenyalcin - do you need anything from me to move this forward?
The `examples/elbv2/lbtargetgroupattachment.yaml` is using the selector.  I don't think I have the ability to trigger the automated tests.
If there's anything else I need for this contribution can you please let me know?

**Comment by sergenyalcin** [comment]
> Hi @sergenyalcin - do you need anything from me to move this forward? The `examples/elbv2/lbtargetgroupattachment.yaml` is using the selector. I don't think I have the ability to trigger the automated tests. If there's anything else I need for this contribution can you please let me know?

As far as I see, the manifest in the `examples-generated` (it is the generated one) uses selector for targetId field. However, manifest in the `examples` folder does not use this selector. We use the manifest in the `examples` folder for automated-test. I meant fixing and adding this. After that I can trigger the automated tests.

**Comment by smcavallo** [comment]
hi @sergenyalcin - I left the existing example with a hard coded instance ID.  I added an example with a `targetIdSelector` which resolves to an ALB.

**Comment by smcavallo** [comment]
@turkenf - all set

File: examples/elbv2/lbtargetgroupattachment.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
 apiVersion: elbv2.aws.upbound.io/v1beta1
 kind: LBTargetGroupAttachment
 metadata:
-  name: test
+  name: test-instance
```
Comment (ulucinar) [inline examples/elbv2/lbtargetgroupattachment.yaml:21]: Could you please add the `meta.upbound.io/example-id` annotation to the manifest:
```suggestion
  name: test-instance
  annotations:
    meta.upbound.io/example-id: elbv2/v1beta1/lbtargetgroupattachment
```

File: examples/elbv2/lbtargetgroupattachment.yaml [hunk 2]
```diff
@@ -29,3 +29,102 @@ spec:
       matchLabels:
         testing.upbound.io/example-name: elbv2
     targetId: 172.16.10.25
+---
+apiVersion: ec2.aws.upbound.io/v1beta1
```
Comment (ulucinar) [inline examples/elbv2/lbtargetgroupattachment.yaml:33]: Could you please add the `meta.upbound.io/example-id: elbv2/v1beta1/lbtargetgroupattachment` annotation to the dependencies also?

**Review by ulucinar — COMMENTED** [review]
Thanks @smcavallo for working on this! Left some comments for you to consider.

File: config/elbv2/config.go [hunk 3]
```diff
@@ -2,13 +2,21 @@ package elbv2
...
 		r.UseAsync = true
 		r.LateInitializer.IgnoredFields = []string{"access_logs"}
 	})
 	p.AddResourceConfigurator("aws_lb_target_group_attachment", func(r *config.Resource) {
+		r.References = config.References{
+			"target_id": config.Reference{
+				Type: "github.com/upbound/provider-aws/apis/elbv2/v1beta1.LB",
+			},
+			"target_group_arn": config.Reference{
+				Type: "github.com/upbound/provider-aws/apis/elbv2/v1beta1.LBTargetGroup",
```
[the remark this answers is not in the document]
Comment (turkenf) [reply config/elbv2/config.go:17]: The `target_group_arn` selector already comes and works without adding the reference-resolver, I think there is no need to add an extra configuration.

**Comment by jeanduplessis** [comment]
@smcavallo I just wanted to check in to see if you are still interested in pushing this PR forward. If not, we'll proceed with closing it.

**Pull request closed without merging**